santosfamilyfoundation/RuinaMain
# Merge/casey j may/master validation merge

# Master Validation Merge Pull Request

## Description
This change is a manual merge between Text Field Validation and Master. The feature should have validation on all text fields, including the text field with selection component. 

## Testing
1. Switch to this branch
2. Type something into one of each type of text field.
3. Close the app.
4. Reload the report you were just working on.
5. Confirm that all the same text remains.
6. Mess around with the fields to make sure they turn red when there is no text, and have no highlight when there is text.
7. Maybe try and export, just to make sure there aren't any shenanigans going on!

## Note:

In the single select text fields, if you close and reload the app the button select will not stay, but the text will stay filled in. This is a known issue, but is going to be on the back burner for now.
